packages/integration-sdk-runtime/src/synchronization/batchBySize.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Entity,
+  IntegrationLogger,
+  Relationship,
+} from '@jupiterone/integration-sdk-core';
+import { UploadDataLookup } from '.';
+import { MAX_BATCH_SIZE } from './shrinkBatchRawData';
+const BATCH_THRESHOLD = 0.9; //will stop chunking if chunk size is between BATCH_THRESHOLD*sizeInBytes and sizeInBytes
+
+export function chunkBySize<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+  sizeInBytes: number,
+  logger: IntegrationLogger,
+): T[K][][] {
+  return chunk(data, sizeInBytes, logger);
+}
+export function chunk<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+  sizeInBytes: number,
+  logger: IntegrationLogger,
+): T[K][][] {
+  const chunkedData: T[K][][] = [];
+  if (getSizeOfObject(data) < sizeInBytes) {
```

Comment:
Still not sure this is quite right - we're now still calculating the size of the entire batch, plus the size of each individual object, so calculating size twice. 

Let's meet and talk through https://github.com/JupiterOne/sdk/pull/925#discussion_r1277783164